### PR TITLE
move REST API related code over from TE

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/70712

Base for: https://github.com/moderntribe/the-events-calendar/pull/1234

This PR moves REST API related code we might want to reuse over in common.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space before the last parenthesis | [Base_Header.php:16](https://github.com/moderntribe/tribe-common/blob/d4d019daeabd17ec10d516e07f36a12f13453bf5/src/Tribe/REST/Headers/Base_Header.php#L16)
:x: | There must be a single space before the last parenthesis | [Base_Header.php:16](https://github.com/moderntribe/tribe-common/blob/d4d019daeabd17ec10d516e07f36a12f13453bf5/src/Tribe/REST/Headers/Base_Header.php#L16)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Supported.php:28](https://github.com/moderntribe/tribe-common/blob/d4d019daeabd17ec10d516e07f36a12f13453bf5/src/Tribe/REST/Headers/Supported.php#L28)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Supported.php:29](https://github.com/moderntribe/tribe-common/blob/d4d019daeabd17ec10d516e07f36a12f13453bf5/src/Tribe/REST/Headers/Supported.php#L29)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Supported.php:30](https://github.com/moderntribe/tribe-common/blob/d4d019daeabd17ec10d516e07f36a12f13453bf5/src/Tribe/REST/Headers/Supported.php#L30)

> via jenkins/codesniffer-phpcs

**Comment by borkweb** [comment]
:1st_place_medal:

**Comment by borkweb** [comment]
Adding `hold` on this until we get 4.4 out the door.

File: src/Tribe/REST/Main.php [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+		/**
+		 * Filters The Events Calendar REST URL.
+		 *
+		 * @param string $url     TEC REST URL.
+		 * @param string $path    REST route.
+		 * @param int    $blog_id Blog ID.
+		 * @param string $scheme  Sanitization scheme.
+		 */
+		return apply_filters( 'tribe_events_rest_url', $url, $path, $blog_id, $scheme );
```
Comment (bordoni) [inline src/Tribe/REST/Main.php]: Remove the `tribe_events` in favor of the Common namespace.

File: src/Tribe/REST/Headers/Disabled.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+	/**
+	 * Sends TEC REST API related headers.
+	 */
+	public function send_header() {
+		if ( headers_sent() ) {
+			return;
+		}
+
+		header( '' . $this->base->get_api_version_header() . ': disabled' );
```
Comment (bordoni) [inline src/Tribe/REST/Headers/Disabled.php]: Why having that initial `''` there

Comment (lucatume) [inline src/Tribe/REST/Headers/Disabled.php]: My mistake

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space after the first parenthesis | [Unsupported.php:17](https://github.com/moderntribe/tribe-common/blob/ecec69aab26d76d16c16cf1a4ce3fb0b43e1c584/src/Tribe/REST/Headers/Unsupported.php#L17)
:x: | There must be a single space after the first parenthesis | [Unsupported.php:17](https://github.com/moderntribe/tribe-common/blob/ecec69aab26d76d16c16cf1a4ce3fb0b43e1c584/src/Tribe/REST/Headers/Unsupported.php#L17)
:x: | There must be a single space after the first parenthesis | [Unsupported.php:18](https://github.com/moderntribe/tribe-common/blob/ecec69aab26d76d16c16cf1a4ce3fb0b43e1c584/src/Tribe/REST/Headers/Unsupported.php#L18)
:x: | There must be a single space before the last parenthesis | [Unsupported.php:18](https://github.com/moderntribe/tribe-common/blob/ecec69aab26d76d16c16cf1a4ce3fb0b43e1c584/src/Tribe/REST/Headers/Unsupported.php#L18)

> via jenkins/codesniffer-phpcs

**Comment by lucatume** [comment]
rebased the PR against `feature/rest-api` branch

**Pull request merged**